src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/EmbeddedLanguages/VirtualChars/CSharpVirtualCharService.cs

```diff
@@ -391,14 +391,14 @@ public override bool TryGetEscapeCharacter(VirtualChar ch, out char escapedChar)
         private static int HexValue(char c)
         {
             Debug.Assert(IsHexDigit(c));
-            return (c >= '0' && c <= '9') ? c - '0' : (c & 0xdf) - 'A' + 10;
+            return (c is >= '0' and <= '9') ? c - '0' : (c & 0xdf) - 'A' + 10;
         }
 
         private static bool IsHexDigit(char c)
         {
-            return (c >= '0' && c <= '9') ||
-                   (c >= 'A' && c <= 'F') ||
-                   (c >= 'a' && c <= 'f');
+            return c is >= '0' and <= '9' or
+                   >= 'A' and <= 'F' or
+                   >= 'a' and <= 'f';
```

Comment:
the parentheses here were good for precednece undestanding.  it's not great that htey got removed.